src/lib/sql/sql_translator.cpp

```diff
@@ -522,8 +525,9 @@ SQLTranslator::TableSourceState SQLTranslator::_translate_table_origin(const hsq
                 "Must specify a name for exactly each column");
 
     for (auto column_id = ColumnID{0}; column_id < hsql_table_ref.alias->columns->size(); ++column_id) {
-      sql_identifier_resolver->set_column_name(column_expressions[column_id],
-                                               (*hsql_table_ref.alias->columns)[column_id]);
+      const auto& expression = column_expressions[column_id];
+      sql_identifier_resolver->reset_column_names(expression);
```

Comment:
Why does it have to be reset?

Reply:
This is especially for queries like `SELECT * FROM int_float AS t (b, a) WHERE b = t.a`, in which the original column names should not be accessible anymore.

Reply:
Please add this as a comment, I think that example helps.

Reply:
I added a shorter version of this comment because there already is an explanation a few lines earlier.